bibernate-core/src/main/java/com/bobocode/petros/bibernate/utils/StatementUtils.java

```diff
@@ -0,0 +1,116 @@
+package com.bobocode.petros.bibernate.utils;
+
+import com.bobocode.petros.bibernate.exceptions.JdbcOperationException;
+import com.bobocode.petros.bibernate.session.query.condition.Restriction;
+import com.bobocode.petros.bibernate.session.query.enums.QueryType;
+import lombok.experimental.UtilityClass;
+
+import java.lang.reflect.Field;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Utility class that operates over {@link PreparedStatement} for the different {@link QueryType}.
+ */
+@UtilityClass
+public class StatementUtils {
+
+    /**
+     * Applies specific logic required for SELECT statement.
+     *
+     * @param statement    instance of {@link PreparedStatement}
+     * @param entityClass  instance of {@link PreparedStatement}
+     * @param restrictions applicable restrictions
+     * @param <T>          generic type
+     * @return {@link PreparedStatement} with all parameters set
+     */
+    public <T> PreparedStatement prepareSelectStatement(final PreparedStatement statement,
+                                                        final Class<T> entityClass,
+                                                        final List<Restriction> restrictions) {
+        if (!restrictions.isEmpty()) {
+            final List<Field> fields = EntityUtils.sortEntityFieldsSkipPK(entityClass);
+            int magicNumber = 1;
+            for (final Restriction restriction : restrictions) {
+                if (fields.stream().anyMatch(f -> f.getName().equals(restriction.getPropertyName()))) {
+                    try {
+                        statement.setObject(magicNumber++, restriction.getValue());
+                    } catch (SQLException e) {
+                        throw new JdbcOperationException(e.getMessage(), e);
+                    }
+                }
+            }
+        }
+        return statement;
+    }
+
+    /**
+     * Applies specific logic required for INSERT statement.
+     *
+     * @param statement instance of {@link PreparedStatement}
+     * @param entity    insertable entity
+     * @param <T>       generic type
+     * @return {@link PreparedStatement} with all parameters set
+     */
+    public <T> PreparedStatement prepareInsertStatement(final PreparedStatement statement, final T entity) {
+        final List<Field> fields = EntityUtils.sortEntityFieldsSkipPK(entity.getClass());
+        int magicNumber = 1;
```

Comment:
please, use more meaningful names for these local variables. they're at least a column numbers